tests/test_java.py

```diff
@@ -653,3 +658,46 @@ def test_java_different_basename(
                     os.path.basename(log_records[0].gprofiler_adapter_extra["exe"])  # type: ignore
                     == java_notjava_basename
                 )
+
+
+@pytest.mark.parametrize("in_container", [True])
+@pytest.mark.parametrize("insert_dso_name", [False, True])
+def test_dso_name_in_ap_profile(
+    tmp_path: Path,
+    application_pid: int,
+    insert_dso_name: bool,
+) -> None:
+    with make_java_profiler(
+        storage_dir=str(tmp_path),
+        insert_dso_name=insert_dso_name,
+        duration=3,
+        frequency=999,
+    ) as profiler:
+        collapsed = snapshot_pid_profile(profiler, application_pid).stacks
+        assert is_function_in_collapsed("jni_NewObject", collapsed)
+        assert insert_dso_name == is_pattern_in_collapsed(r"jni_NewObject \(.+?/libjvm.so\)", collapsed)
+
+
+# test that occurrence of raw DSO path as a symbol is recognized as such by async profiler
+@pytest.mark.parametrize("in_container", [True])
+@pytest.mark.parametrize(
+    "insert_dso_name, libc_pattern",
+    [
+        (False, r"(^|;)\(/.*/libc-.*\.so\)($|;)"),
+        (True, r"(^|;)\[unknown\] \(/.*/libc-.*\.so\)($|;)"),
```

Comment:
Isn't it like this case? https://github.com/Granulate/gprofiler/pull/491/files#diff-67815cf6790347c8dd837878986a4d4b2dcec6eddc15ebb573ca0a408fe5a778R53

So I expect just the DSO in the `True` case as well, because the symbol is unknown.

Reply:
Yes, exactly.
So it will be just the library name with or without DSO. Got it.